.github/workflows/python-package.yml

```diff
@@ -5,47 +5,12 @@ name: Python package
 
 on:
   push:
-    branches: [ main, develop, v0.2-staging ]
+    branches: [ main, develop ]
   pull_request:
-    branches: [ main, develop, v0.2-staging ]
+    branches: [ main, develop ]
 
 jobs:
 
-  build38:
-
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
-        python-version: ["3.8"]
-        numpy-version: ["1.18", "1.19"]
-        scipy-version: ["1.4"]
-    steps:
-    - uses: actions/checkout@v4
-    - uses: actions/setup-node@v4
-      with:
-        node-version: 20
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
-      with:
-        python-version: ${{ matrix.python-version }}
-        cache: 'pip'
-    - name: Install dependencies
-      run: |
-        pip install --upgrade pip setuptools Cython
-        pip install numpy==${{ matrix.numpy-version }}
-        pip install scipy==${{ matrix.scipy-version }}
-        pip install -r requirements.txt
-        pip install -e .
-        pip install flake8 pytest pytest-cov
-    - name: Test with flake8
-      run: |
-        flake8
-    - name: Test with pytest
-      run: |
-        ./tests/run-tests.sh
-
-
   build39_310_311:
```

Comment:
Better to decouple generic action text from specific versions?  How about `build_supported` with the idea that we could add in a `build_312` as you test correctness with 3.12 and then phase that out when 3.12 is added to `build_supported`?

Out of curiosity, do you really want to test with fixed versions of numpy and scipy?  In the present implementation, we either won't detect that surmise doesn't work with newer versions or we have a new maintenance task of remembering to update versions every so often.

Related to this, we could eventually setup your repo to use dependabot
https://docs.github.com/en/code-security/getting-started/dependabot-quickstart-guide

Reply:
Separating from specific versions sounds like a great idea. I agree with the `build_supported` and `build_312`.

Practically speaking, we probably wouldn't care which versions of numpy/scipy until something breaks.  *Or*, we keep track of the last versions that work? 

A dependabot implementation would be good.

Reply:
I believe that when a new version of `numpy` is released, for example, dependabot creates a new PR as a "bump."  This would then trigger your PR-based CI actions.
https://github.com/Libensemble/libensemble/pull/1319

 If you see that all those actions have passed, then you assume that your code is compatible with the new version and you either merge or close the PR.  I would, therefore, believe that your CI actions need to use the latest version of dependencies.

If there is a breakage, ideally you would fix your software ASAP.  If you can't do so quickly, then we would have to figure out how to temporarily manage the breakage by, for instance, putting an upper limit on versions in setup.py or a requirements file.  I've never done that, but assume that you can do that.

I haven't done this yet, but on other projects I have the goal of setting up an action that tests the oldest versions of dependencies as specified in your setup.py or requirements files.  In that way, the CI would be bookending the set of supported versions.

Reply:
This makes sense. I actually just encountered this issue elsewhere since `numpy 2.0.0` is released and PyTorch can have compatibility issues.

In requirements, the syntax would be, e.g., `numpy>=1.18.3,<2`.

I like the idea of testing the bookending versions.